modules/ROOT/pages/ch2-api.adoc

```diff
@@ -0,0 +1,376 @@
+** Manage static IP addresses and release unused IP addresses associated with an application.
+** Download the Mule package file for the application.
+** Update, enable, disable, or run all or some schedules.
+
+
+* Manage logs, notifications, and alerts:
+** Add, update, or retrieve log levels for an application.
+** Download the log file for an application or instance.
+** Create, update, or retrieve a notification.
+** Mark a notification as read or unread.
+** Create, update, or delete an alert triggered by an application.
+** Retrieve an alert and alert history.
+
+* Manage load balancers:
+** Create, update, or delete a load balancer.
+** List load balancers in an organization.
+** Test load balancer rules and return the number of the rule applied to the input.
+
+* Manage private spaces, VPNs, and transit gateway attachments:
+** Create, update, list, and remove a VPC.
+** Create or delete a VPN connection.
+** Create or delete a transit gateway attachment.
+
+
+* Retrieve additional information about:
+** Supported Mule versions
+** The current organization, including usage statistics
+** Replicas, including thread dumps and regions
+** Users, including permissions
+** Available domain names
+
+////
+
+
+For an interactive reference that includes supported resources, methods, required properties, and expected responses, see the https://anypoint.mulesoft.com/exchange/portals/anypoint-platform/f1e97bc6-315a-4490-82a7-23abe036327a.anypoint-platform/cloudhub-api/[CloudHub 2.0 API^] (Link out from Beta docs).
+
+<update link>
+
+
+The CloudHub 2.0 API manages only applications deployed to CloudHub 2.0.
+To manage on-premises applications using the API, see https://anypoint.mulesoft.com/exchange/portals/anypoint-platform/f1e97bc6-315a-4490-82a7-23abe036327a.anypoint-platform/arm-rest-services[Runtime Manager REST Services^] (Link out from Beta docs).
+
+
+
+== Use the CloudHub 2.0 API
+
+Before getting started, familiarize yourself with operations for applications.
+
+You can use any HTTP client with the CloudHub 2.0 API.
+With Java, use the https://jersey.github.io/[Jersey client^] or http://hc.apache.org/httpclient-3.x/index.html[HttpClient^] with https://github.com/codehaus/jackson[Jackson^] for JSON support. (Links out from Beta docs)
+
+
+=== Authenticate with the CloudHub 2.0 APIs
+
+To access operations in the CloudHub 2.0 API, you must:
+
+include::partial$ch2-api-authenticate.adoc[]
+
+
+=== Deploy an App
+
+To deploy an app to CloudHub 2.0, run the following `curl` command:
+
+----
+curl -X GET \
+
+Need example code here
+
+----
+
+*Martin?*
+
+=== Modify an Attribute in a Private Space
+
+
+To modify an attribute in a private space, run the following `curl` command:
+
+----
+curl -X GET \
+Need example code here
+
+----
+
+*Xuan?*
+
+
+=== Add a Firewall Rule 
+
+
+To add a firewall rule to a private space, run the following `curl` command:
+
+----
+curl -X GET \
+Need example code here
```

Comment:
```
curl --request PATCH 'https://anypoint.mulesoft.com/runtimefabric/api/organizations/<org-id>/privatespaces/<private-space-id>' \
--header 'Content-Type: application/json' \
--header 'Authorization: <bearer token>' \
--data-raw '{
"managedFirewallRules": [
        {
            "cidrBlock": "0.0.0.0/0",
            "protocol": "tcp",
            "fromPort": 443,
            "toPort": 443,
            "type": "inbound"
        }
}'
```

Reply:
Note, you need to provide all the rules in the `managedFirewallRules` property, not just the new ones you wish to add.

Reply:
Added